05/DB.test.js

```diff
@@ -1 +1,149 @@
-import DB from './DB';
\ No newline at end of file
+import DB from "./DB";
+
+describe("Database", () => {
+  describe("Insert item", () => {
+    it("rejects when given id is not a number", async () => {
+      const database = new DB();
+      const data = { id: "ID", a: 10 };
+
+      await expect(database.insert(data)).rejects.toEqual(
+        "ID can be only number!"
+      );
+      expect(database._rows).not.toContainEqual(data);
+    });
+
+    it("rejects when given id is not unique", async () => {
+      const database = new DB();
+      database._rows = [{ id: 1, a: 10 }];
+      const data = { id: 1, a: 20 };
+
+      await expect(database.insert(data)).rejects.toEqual(
+        "ID can't be duplicated!"
+      );
+      expect(database._rows).toEqual([{ id: 1, a: 10 }]);
+    });
+
+    it("resolves and creates new record when id is valid", async () => {
+      const database = new DB();
+      const data = { id: 1, a: 10 };
+
+      await expect(database.insert(data)).resolves.toStrictEqual(data);
+      expect(database._rows).toContainEqual(data);
+    });
+
+    it("resolves and creates new record with unique id when id is not provided", async () => {
+      const database = new DB();
+      database._rows = [{ id: 1, a: 10 }];
```

Comment:
Raczej nie operujemy na prywatnych polach, znak podkreślenia (_) raczej świadczy o tym, że takie jest to pole.
Dlatego lepiej wykorzystać metodę, która doda dane i możliwe, że wykona jakieś dodatkowe działanie, które zostanie pominięte jeśli od razu przypiszemy dane do `_rows`.